RoboJackets/robocup-rustware
# Fix FPGA Initialization Issues

# Overview

Currently receiving radio commands can interrupt the initialization of the FPGA.  This isn't great because it makes us turn off the base station every time a robot is turned on and off.  This PR places the FPGA initialization into a critical section and only turns on the radio receive interrupt once the FPGA has been initialized.

This PR also cleans up a ton of logic and makes it standard to initialize drivers in critical-sections of software tasks.

## Steps to Test

1. (✅) Flash the robot with manual control
2. (✅) Run the base station
3. (✅) Turn on the robot and ensure the fpga initializes correctly (the on-board led should go to about 50% brightness).
4. (✅) Send move commands from the base station telling the robot to move and observe the wheels turning.

## Linked Clickup Tasks
[Fix Panic from Radio Interrupt During FPGA Initialization](https://app.clickup.com/t/86b1vtcr2)
